BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -861,6 +863,17 @@ PciScanBus (
         Address   = PCI_EXPRESS_LIB_ADDRESS (StartBusNumber, Device, Func, PCI_BRIDGE_SUBORDINATE_BUS_REGISTER_OFFSET);
         PciExpressWrite8 (Address, (UINT8)Register);
 
+        //
+        // Some controllers require read-write 0xffffffff to be detected.
+        // Perform this in Hook function.
+        //
+        Address   = PCI_EXPRESS_LIB_ADDRESS (StartBusNumber, Device, Func, PCI_BRIDGE_SECONDARY_BUS_REGISTER_OFFSET);
+        Register  = (UINT8)PciExpressRead8 (Address);
+        PlatformPciEnumHook = (PLATFORM_PCI_ENUM_HOOK)(UINTN)PcdGet32 (PcdFuncPciEnumHook);
+        if ((Register != 0) && (PlatformPciEnumHook != NULL)) {
+          PlatformPciEnumHook ((UINT32)Register);

```

Comment:
Should we use a more generic function similar to EDK2 PCI bus driver?
PlatformPciEnumHookProc (Phase, PciAddress) ?